src/compute/src/compute_state.rs

```diff
@@ -501,15 +501,12 @@ impl<'a, A: Allocate + 'static> ActiveComputeState<'a, A> {
 
         let (logger, traces) = logging::initialize(self.timely_worker, config);
 
-        // Install traces as maintained indexes
-        for (log, trace) in traces {
+        for (log, (trace, dataflow_index)) in traces {
+            // Install trace as maintained index.
             let id = config.index_logs[&log];
             self.compute_state.traces.set(id, trace);
-        }
 
-        // Initialize compute and logging state for each logging index.
-        let index_ids = config.index_logs.values().copied();
-        for id in index_ids {
+            // Initialize compute and logging state for the logging index.
```

Comment:
This removes an implicit check that we have a log for all `config.index_logs`. If it's something we want to keep, we should make it explicit!

Reply:
I was wondering about this as well. The previous code would make sure that a `CollectionState` and logging was added for all logs in the config, but it didn't actually check to ensure that we also have a trace for each of them. That doesn't seem very useful: I'm pretty sure if anyone would actually want to query a collection that has no known trace, we would just panic anyway.

Probably doesn't hurt to add a check here that ensures all requested variants have a trace though.

Reply:
Added a sanity check.